src/index/IndexImpl.cpp

```diff
@@ -959,11 +960,13 @@ void IndexImpl::createFromOnDiskIndex(const std::string& onDiskBase,
     load(osp_);
     load(spo_);
     load(sop_);
+    setOriginalMetadataFor(pso_, pos_, ops_, osp_, spo_, sop_);
   } else {
     AD_LOG_INFO
         << "Only the PSO and POS permutation were loaded, SPARQL queries "
            "with predicate variables will therefore not work"
         << std::endl;
+    setOriginalMetadataFor(pso_, pos_);
```

Comment:
What's the difference with this change? To me it looks like it does the same thing as before but with variadic templates and a second function call?

Reply:
The difference is that the metadata is loaded for all permutations using only a single call to `DeltaTriplesManager::modify`. I did my best to explain it in the updated comment above. The Metadata Update logs a warning if a permutation does not have the original metadata available. We now get warnings that are to be expected during the Index load, because the Metadata is updated after the modify when setting the metadata after loading a permutation. We can also add another flag to modify for disabling the metadata update, if you prefer that.